frontend-ssr/pages/api/user/login.js

```diff
@@ -0,0 +1,38 @@
+//Imports
+const User = require("../../../models/User.schema");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const { loginValidation } = require("../../../utils/validation/userValidation")
+
+//Vars
+const error_message = { status: "fail", message: 'Email or Password is incorrect' }
+
+export default async (req, res) => {
+    let { method, body } = { ...req };
+    switch (method) {
+        case "POST":
+            //destruct the req
+            let { email, password } = { ...body };
+            let { error } = loginValidation({ email, password })
+            //Check valid form 
+            if (error) return res.status(400).json(error_message);
+            //Check user base on username or email
+            let user = await User.findOne({ $or: [{ username: email }, { email: email }] });
```

Comment:
What if this one fails? We should still return a 500 message with something like "Oups, something went wrong. Try again.". 

Question: do we have to do this for all calls to the DB or can we implement a generic catch for fails like this?

Reply:
That's a good question, I will look into it for a generic error handling for DB connection.